docs/admin-manual/configuration-macros.rst

```diff
@@ -3777,6 +3777,12 @@ prevent the job from using more scratch space than provisioned.
     will be marked as broken if the associated ephemeral logical volume
     is failed to be cleaned up.
 
+:macro-def:`NO_JOB_NETWORKING[STARTD]`
+    A boolean expression that, when ``True``, disables access to the 
+    network by a job.  Only jobs that opt into such machines will 
+    match and run on such a machine.
+
+
```

Comment:
This sounds like it can be an expression that's evaluated in the context of the slot and/or job ads.